addons/web/static/src/js/core/owl_dialog.js

```diff
@@ -0,0 +1,248 @@
+        }
+        return [ DEFAULT_BUTTON ];
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Closes and unmounts the dialog.
+     *
+     * @param {Object} [options={}]
+     * @param {Object} [options.closedBy] source of the close call
+     * @param {Object} [options.infos] if provided and `silent` is unset, the
+     *   `on_close` handler will pass this information related to closing this
+     *   information.
+     * @param {boolean} [options.silent=false] if set, do not call the
+     *   `on_close` handler.
+     */
+    close(options={}) {
+        this.trigger('dialog_closed', Object.assign({
+            closedBy: null,
+        }, options));
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _removeTooltips() {
+        [...document.querySelectorAll('.tooltip')].forEach(tooltip => {
+            tooltip.parentNode.removeChild(tooltip); // remove open tooltip if any to prevent them staying when modal is opened
+        });
+    }
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {number} buttonIndex
+     */
+    _onButtonClick(buttonIndex) {
+        const button = this.props.buttons[buttonIndex];
+        this.trigger('dialog_button_clicked', { button });
+    }
+    /**
+     * @private
+     */
+    _onCloseDialog() {
+        this.close();
+    }
+    /**
+     * Manages the TAB key on the buttons. If you the focus is on a primary
+     * button and the users tries to tab to go to the next button, display
+     * a tooltip
+     *
+     * @private
+     * @param {KeyboardEvent} ev
+     */
+    _onFooterButtonKeyDown(ev) {
+        switch (ev.key) {
+            case 'Tab':
+                if (!ev.shiftKey && ev.target.classList.contains("btn-primary")) {
+                    ev.preventDefault();
+                    const primaryButton = ev.target;
+                    $(primaryButton).tooltip({
+                        delay: { show: 200, hide: 0 },
+                        title: () => qweb.render('FormButton.tooltip', { title: primaryButton.innerText.toUpperCase() }),
+                        trigger: 'manual',
+                    });
+                    $(primaryButton).tooltip('show');
+                }
+        }
+    }
+}
+
+OwlDialog.template = 'OwlDialog';
+OwlDialog.defaultProps = {
+    backdrop: 'static',
+    buttons: [],
+    dialogClass: '',
+    fullscreen: false,
+    onForceClose: false,
+    renderFooter: true,
+    renderHeader: true,
+    sizeClass: 'large',
+    subComponent: false,
```

Comment:
remove -> type should always be 'owl.Component' or key absent

Reply:
Ok